File: Solutions/Solution55.cs
public class Solution {
    public bool CanJump(int[] nums) {
        int target = nums.Length - 1;
        
        for (int i = nums.Length - 1; i >= 0; i--) 
        {
            if (i + nums[i] >= target)
            {
                target = i;
            }
        }

        return target == 0;
    }
}

File: Solutions/Solution55.rs
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut target = nums.len() - 1;

        for index in (0..nums.len()).rev() {
            if (index + (nums[index] as usize) >= target) {
                target = index;
            }
        }

        target < 1
    }
}